### DISCUSSION TITLE
Flutter MacOS code signing

### DISCUSSION THREAD

**Discussion by GmeWrks21** [codemagic-ci-cd/codemagic-docs#1754] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Add a flutter workflow with workflow editor.
2. Setup the workflow only to build and distribute for MacOS.
3. Use manual code signing don't use AppStore connect(cause getting errors there also don't want to publish to AppStore). 
4. Start a build.

### Expected results

Be able to open the .pkg file/ extract the .zip file and use the app without any problems.

### Actual results

The built .pkg can't be opened because the Gatekeeper stops saying that this app is from an unidentified developer also getting this from the build logs,
"warning: Runner isn't code signed but requires entitlements. It is not possible to add entitlements to a binary without signing it. (in target 'Runner' from project 'Runner')"

### Build id (optional)

648629d83e15a9344b3ef705

**Comment by himesh-cm** [comment]
Hi @GmeWrks21,

You need to allow the application to open; you can check the links [here](https://support.apple.com/en-in/guide/mac-help/mh40616/mac) & [here](https://support.apple.com/en-in/HT202491) for more details.

Please try and let me know.

**Comment by GmeWrks21** [comment]
@himesh-cm thanks for the response but it isn't great asking people to do that. How can I avoid the gatekeeper? I mean how can I get a verified pkg?

**Reply by himesh-cm** [reply]
@GmeWrks21, How do you plan to publish the app? Is it via the app-store or directly from a download link?
Please note your team is on a paid plan, and you should have access to the intercom chat widget. Please connect us via intercom for faster replies.

**Reply by IsmailAlamKhan** [reply]
Directly from a download link. Alright I will do that but for now I see that Codemagic doesn't support signing with  DeveloperID.

**Reply by himesh-cm** [reply]
Yes, For workflow Editor, Currently, Codemagic only supports code signing the app for publishing to the Mac App Store.
You can use the `yaml` approach, which supports `DeveloperID` as well; please check the document [here](https://docs.codemagic.io/yaml-code-signing/signing-macos/)

**Reply by himesh-cm** [reply]
Hi @IsmailAlamKhan, Did you get a chance to try the `yaml` configuration?

**Reply by IsmailAlamKhan** [reply]
> Hi @IsmailAlamKhan, Did you get a chance to try the `yaml` configuration?

No yet to try currently on a vacation

**Reply by himesh-cm** [reply]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
